R/convertToSE.R

```diff
@@ -0,0 +1,40 @@
+#' It helps to store any matrix in \linkS4class{SummarizedExperiment}
+#' object class data container(`SE`). The resulting abundance matrix from the
+#' simulation functions used in `miaSim` can be used.
+#'
+#' @param matrix : A matrix-like or list of matrix-like objects.
+#' Rows: genes, genomic coordinates, etc. Columns: samples, cells, etc.
```

Comment:
https://www.bioconductor.org/help/course-materials/2019/BSS2019/04_Practical_CoreApproachesInBioconductor.html

It was mentioned here. It can be deleted if it is unnecessary.

Reply:
I didn't borrow the function from another package but if needed it can be replaced to this function that is used in SE package itself, [makeSummarizedExperimentFromDataFrame](https://rdrr.io/bioc/SummarizedExperiment/man/makeSummarizedExperimentFromDataFrame.html). It uses data frame instead of matrix and the end result is RangedSummarizedExperiment.

Reply:
Oke - in miaverse context we are mainly discussing microbial taxa at the moment, not genes or genomes (although this might come into play later). Perhaps most clear to refer to "taxa" and "samples" for now, for rows and cols.

If `makeSummarizedExperimentFromDataFrame` works, do we have a particular need to have our own function `convertToSE`? What is the added value? In general, we should aim to reuse existing tools from other packages as much as possible, and only reimplement if we can do it better or if we need something genuinely different.

Reply:
I agree, here the differences in using `makeSummarizedExperimentFromDataFrame` are that it returns `RangedSummarizedExperiment`  and output matrix from simulations in the package cannot be applied directly because it requires data frame. Also, for some reason, `as.data.frame(matrix)` still cannot be applied. It throws the error `Error in .find_start_end_cols(df_colnames0, start.field0, end.field0) : 
  cannnot determine start/end columns`